beacon8/layers/Overfeat.py

```diff
@@ -0,0 +1,246 @@
+from . import Module
+
+
+import theano as _th
+from theano.sandbox.cuda import CudaNdarrayType, GpuOp
+from theano.sandbox.cuda.basic_ops import (as_cuda_ndarray_variable, gpu_contiguous)
+
+
+class PyCudaOp(GpuOp):
+    def __eq__(self, other):
+        return type(self) == type(other)
+
+    def __hash__(self):
+        return hash(type(self))
+
+    def __str__(self):
+        return self.__class__.__name__
+
+    def make_node(self, inp):
+        inp = as_cuda_ndarray_variable(inp)
+        return _th.Apply(self, [inp], [inp.type()])
+
+
+class RollOpBase(PyCudaOp):
+    def c_support_code(self):
+        c_support_code = """
+            __global__ void maxpool_roll(float *input, float *output, int batch_size, int feature_size, int height_size, int width_size)
+        {
+            int x = blockIdx.x * blockDim.x + threadIdx.x;
+            int batch = blockIdx.y * blockDim.y + threadIdx.y;
+            int map_size = height_size * width_size;
+            int feature = x / map_size;
+            int height = (x % map_size) / width_size;
+            int width = x % width_size;
+            int height_out = height / 2;
+            int width_out = width / 2;
+            int batch_out = batch * 4;
+            if (height % 2 == 0 && width % 2 == 1)
+            {
+                batch_out += 1;
+            }
+            else if (height % 2 == 1 && width % 2 == 0)
+            {
+                batch_out += 2;
+            }
+            else if (height % 2 == 1 && width % 2 == 1)
+            {
+                batch_out += 3;
+            }
```

Comment:
I think we could write the three above as `batch_out += 2*(height % 2) + (width % 2)`, which would avoid conditionals and thus, in theory, speed up this kernel quite a bit.